spirv_hlsl.cpp

```diff
@@ -1798,6 +1850,199 @@ void CompilerHLSL::emit_glsl_op(uint32_t result_type, uint32_t id, uint32_t eop,
 	}
 }
 
+string CompilerHLSL::read_access_chain(const SPIRAccessChain &chain)
+{
+	auto &type = get<SPIRType>(chain.basetype);
+
+	SPIRType target_type;
+	target_type.basetype = SPIRType::UInt;
+	target_type.vecsize = type.vecsize;
+	target_type.columns = type.columns;
+
+	// FIXME: Transposition?
+	if (type.columns != 1)
+		SPIRV_CROSS_THROW("Reading matrices from ByteAddressBuffer not yet supported.");
+
+	if (type.basetype == SPIRType::Struct)
+		SPIRV_CROSS_THROW("Reading structs from ByteAddressBuffer not yet supported.");
+
+	if (type.width != 32)
+		SPIRV_CROSS_THROW("Reading types other than 32-bit from ByteAddressBuffer not yet supported.");
+
+	const char *load_op = nullptr;
+	switch (type.vecsize)
+	{
+	case 1:
+		load_op = "Load";
+		break;
+	case 2:
+		load_op = "Load2";
+		break;
+	case 3:
+		load_op = "Load3";
+		break;
+	case 4:
+		load_op = "Load4";
+		break;
+	default:
+		SPIRV_CROSS_THROW("Unknown vector size.");
+	}
+
+	auto load_expr = join(chain.base, ".", load_op, "(", chain.dynamic_index, chain.static_index, ")");
```

Comment:
Is a + missing between dynamic_index and static_index here? The same happens in the handling of store.